### PR TITLE
Release v0.57.2

### LINKED ISSUE #1016: Release the operator v0.57.2

### PR DESCRIPTION
**Pull request by yuriolisa**
Signed-off-by: Yuri Sa <yurimsa@gmail.com>
Resolve #1016

### CODE REVIEW

**Comment by yuriolisa** [comment]
@pavolloffay, @jpkrohling

File: config/manager/kustomization.yaml [hunk 1]
```diff
@@ -1,2 +1,8 @@
 resources:
-- manager.yaml
\ No newline at end of file
+- manager.yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
```
Comment (jpkrohling) [inline config/manager/kustomization.yaml:3]: This shouldn't be here, should it?

Comment (yuriolisa) [reply config/manager/kustomization.yaml]: Removed.

**Review by pavolloffay — APPROVED** [review]

**Comment by pavolloffay** [comment]
The health check changed in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/10853 for 0.57.2 release in the contrib.

the operator should adapt:
* automatically upgrade config to the new form
* properly derive HC port/endpoint from the supplied config - https://github.com/open-telemetry/opentelemetry-operator/blob/22c3ccf578e67b4b22df98513dcf2fc97414802e/pkg/collector/adapters/config_to_probe.go#L51

**Comment by yuriolisa** [comment]
@jpkrohling, could you please have a look?

**Comment by VineethReddy02** [comment]
@yuriolisa any specific reason why we are bumping the operator to `0.57.2` from `0.56.0`? I see the issue by Juca, but do not see context on this.

**Comment by yuriolisa** [comment]
> @yuriolisa any specific reason why we are bumping the operator to `0.57.2` from `0.56.0`? I see the issue by Juca, but do not see context on this.

@VineethReddy02, due to the contrib release like @pavolloffay mentioned previously.

**Comment by VineethReddy02** [comment]
@yuriolisa can you fix the unit test that's failing?

**Comment by yuriolisa** [comment]
> @yuriolisa can you fix the unit test that's failing?

It's fixed. Could you please review it? 
@VineethReddy02, @jpkrohling, @jpkrohling.

File: pkg/collector/upgrade/v0_57_0.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+					otelCfg["extensions"] = extensionsConfig
+					res, err := yaml.Marshal(otelCfg)
+					if err != nil {
+						return otelcol, fmt.Errorf("couldn't upgrade to v0.57.2, failed to marshall back configuration: %w", err)
+					}
+
+					otelcol.Spec.Config = string(res)
+					//trings.ReplaceAll(string(res), " null", "")
+					existing := &corev1.ConfigMap{}
+					updated := existing.DeepCopy()
```
Comment (pavolloffay) [inline pkg/collector/upgrade/v0_57_0.go:65]: can you explain lines 64 and 65? Why are you creating a config map, then creating a copy and then using the copy as an object to create an event on? 

Why are we not creating the event on the otelcol CR?

File: pkg/collector/upgrade/v0_57_0.go [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+	if len(otelcol.Spec.Config) == 0 {
+		return otelcol, nil
+	}
+
+	otelCfg, err := adapters.ConfigFromString(otelcol.Spec.Config)
+	if err != nil {
+		return otelcol, fmt.Errorf("couldn't upgrade to v0.57.2, failed to parse configuration: %w", err)
+	}
+
+	//Remove externsions.healthcheckextension - Remove deprecated port field from config. (#12668)
```
[the remark this answers is not in the document]
Comment (yuriolisa) [reply pkg/collector/upgrade/v0_57_0.go]: Added

File: pkg/collector/upgrade/v0_57_0.go [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+					delete(extensions, "port")
+
+					otelCfg["extensions"] = extensionsConfig
+					res, err := yaml.Marshal(otelCfg)
+					if err != nil {
+						return otelcol, fmt.Errorf("couldn't upgrade to v0.57.2, failed to marshall back configuration: %w", err)
+					}
+
+					otelcol.Spec.Config = string(res)
+					//trings.ReplaceAll(string(res), " null", "")
```
[the remark this answers is not in the document]
Comment (yuriolisa) [reply pkg/collector/upgrade/v0_57_0.go]: I forgot to remove it.

File: pkg/collector/upgrade/v0_57_0.go (hunk 2 above)
Comment (yuriolisa) [reply pkg/collector/upgrade/v0_57_0.go]: Right, it's better to post the otelcol. The first idea was to replicate a configmap.

File: pkg/collector/upgrade/versions.go [hunk 5]
```diff
@@ -77,6 +77,10 @@ var (
 			Version: *semver.MustParse("0.56.0"),
 			upgrade: upgrade0_56_0,
 		},
+		{
+			Version: *semver.MustParse("0.57.0"),
```
Comment (VineethReddy02) [inline pkg/collector/upgrade/versions.go:81]: The release version across the PR is specified as `0.57.2`, any specific reason why we are adding it here as `0.57.0`?

**Review by VineethReddy02 — CHANGES_REQUESTED** [review]

File: pkg/collector/upgrade/versions.go (hunk 5 above)
Comment (yuriolisa) [reply pkg/collector/upgrade/versions.go]: I mixed up the versions.

File: pkg/collector/upgrade/v0_57_2.go [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+	otelCfg, err := adapters.ConfigFromString(otelcol.Spec.Config)
+	if err != nil {
+		return otelcol, fmt.Errorf("couldn't upgrade to v0.57.2, failed to parse configuration: %w", err)
+	}
+
+	//Remove deprecated port field from config. (https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/10853)
+	extensionsConfig, ok := otelCfg["extensions"].(map[interface{}]interface{})
+	if !ok {
+		// In case there is no health check config.
```
Comment (jpkrohling) [inline pkg/collector/upgrade/v0_57_2.go:41]:
```suggestion
		// In case there is no extensions config.
```

**Review by jpkrohling — APPROVED** [review]
Just a small nit, LGTM.

**Review by VineethReddy02 — APPROVED** [review]

**Pull request merged**